docs/best-practices.md

```diff
@@ -238,6 +236,31 @@ useful server-side features that might be useful for most of the use cases.
    .serve(":8080", service)
  ```
 
+### Finagle Filters vs. Finch Endpoints
+
+Finch endpoints are designed to be able to substitute (when it's reasonable) both Finagle
+services and filters. While it's totally clear that an `Endpoint` in Finch is a core
+abstraction and it should be used whenever a `Service` is used in Finagle, it's not so obvious
+about filters vs. endpoints.
+
+Due to the `Output` ADT, a Finch endpoint might easily simulate a Finagle filter and *reject* a
+request by responding `Output.failure`. Although, this doesn't completely mean endpoints should
+be preferred to filters in all the cases.
+
+The following rule of thumb might be used to determine what building block (filter or endpoint) to
+pick for a given use case.
+
+Use `Filter` instead of `Endpoint[HNil]` (only `Endpoint[HNil]` might be replaced with a filter)
+when:
+
+- It implements the logic that might be shared across all the endpoints in the program
+  (e.g., authorization, CORS).
+- It only used for side-effects (e.g., logging, metrics).
+
+Please note that "error handling" is a special case in Finch. While it seems like a shared logic
+that might be placed into a filter, it's preferred to use `Endpoint.handle` that allows to convert
+exceptions into `Output.failure`s in the fine grained way.
```

Comment:
I’d probably also add something about when you might actually *want* to handle errors using a `Filter` (as you say it’s usually not a good idea). Something like “Note that if you have a `Filter` that rejects incoming requests (say an authentication filter) you may need to handle exceptions within a filter. This is because your filter will run before the Finch service, so the Finch error handler will never see the exception. Something like Finagle's `com.twitter.finagle.http.filter.ExceptionFilter` for example."